x-pack/legacy/plugins/maps/public/layers/vector_tile_layer.js

```diff
@@ -0,0 +1,225 @@
+    if (sourceDataRequest) {//data is immmutable
+      return;
+    }
+    const requestToken = Symbol(`layer-source-refresh:${ this.getId()} - source`);
+    startLoading(SOURCE_DATA_ID_ORIGIN, requestToken, dataFilters);
+    try {
+      const styleAndSprites = await this._source.getVectorStyleSheetAndSpriteMeta(isRetina());
+      stopLoading(SOURCE_DATA_ID_ORIGIN, requestToken, styleAndSprites, {});
+    } catch(error) {
+      onLoadError(SOURCE_DATA_ID_ORIGIN, requestToken, error.message);
+    }
+  }
+
+  _generateMbId(name) {
+    return this.getId() + '_' + name;
+  }
+
+  _getVectorStyle() {
+    const sourceDataRequest = this.getSourceDataRequest();
+    if (!sourceDataRequest) {
+      return null;
+    }
+    const vectorStyleAndSprites = sourceDataRequest.getData();
+    if (!vectorStyleAndSprites) {
+      return null;
+    }
+    return vectorStyleAndSprites.vectorStyleSheet;
+  }
+
+  _getSpriteMeta() {
+    const sourceDataRequest = this.getSourceDataRequest();
+    if (!sourceDataRequest) {
+      return null;
+    }
+    const vectorStyleAndSprites = sourceDataRequest.getData();
+    return vectorStyleAndSprites.spriteMeta;
+  }
+
+  getMbLayerIds() {
+    const vectorStyle = this._getVectorStyle();
+    if (!vectorStyle) {
+      return [];
+    }
+    return vectorStyle.layers.map(layer => this._generateMbId(layer.id));
+  }
+
+  getMbSourceIds() {
+    const vectorStyle = this._getVectorStyle();
+    if (!vectorStyle) {
+      return [];
+    }
+    const sourceIds = Object.keys(vectorStyle.sources);
+    return sourceIds.map(sourceId => this._generateMbId(sourceId));
+  }
+
+  ownsMbLayerId(mbLayerId) {
+    //todo optimize: do not create temp array
+    const mbLayerIds = this.getMbLayerIds();
+    return mbLayerIds.indexOf(mbLayerId) >= 0;
+  }
+
+  ownsMbSourceId(mbSourceId) {
+    //todo optimize: do not create temp array
+    const mbSourceIds = this.getMbSourceIds();
+    return mbSourceIds.indexOf(mbSourceId) >= 0;
+  }
+
+  _makeNamespacedImageId(imageId) {
+    const prefix = this._source.getSpriteNamespacePrefix() + '/';
+    return prefix + imageId;
+  }
+
+  syncLayerWithMB(mbMap) {
+
+    const vectorStyle = this._getVectorStyle();
+    if (!vectorStyle) {
+      return;
+    }
+
+    let initialBootstrapCompleted = false;
+
+    //sync sources
+    const sourceIds = Object.keys(vectorStyle.sources);
+    sourceIds.forEach(sourceId => {
+      if (initialBootstrapCompleted) {
+        return;
+      }
+      const mbSourceId = this._generateMbId(sourceId);
+      const mbSource = mbMap.getSource(mbSourceId);
+      if (mbSource) {
+        //if a single source is present, the layer already has bootstrapped with the mbMap
+        initialBootstrapCompleted = true;
+        return;
+      }
+      mbMap.addSource(mbSourceId, vectorStyle.sources[sourceId]);
+    });
+
+    if (!initialBootstrapCompleted) {
+
+      //sync spritesheet
+      const spriteMeta = this._getSpriteMeta();
```

Comment:
Since `spriteMeta` could be null, you might want to add a conditional return statement right after this call.